.fixtures.yml

```diff
@@ -4,6 +4,7 @@ fixtures:
       repo: 'git://github.com/puppetlabs/puppetlabs-apt'
     stdlib:
       repo: 'git://github.com/puppetlabs/puppetlabs-stdlib'
+      ref: "4.25.0"
```

Comment:
please don't bump it here. We always want to test against the master branch to catch unreleased errors early.

Reply:
I removed it from the commit even though I don't understand the reason.
Which unreleased errors? From the stdlib library? That would not make much sense...

I can only see downsides on doing this: you are testing with code that it will be *different* from what the actual production code will be based on. How does that make sense?

Reply:
that's a complicated story. In the past our modules broke often because a breaking change was introduced into a dependency, back then we references releases in the .fixtures.yml. We pinned it to the lowest known working version. The puppet module tool always tries to download the latest version for dependencies.

tldr: We noticed that dependencies aren't compatible in the latest version only because it failed for our users and they reported it.

What we now do is a bit smarter. We always test against the master branches of our dependencies. So we notice during our regular PRs to our modules, if a dependency isn't working anymore. This is all related to our unit tests.

-----

Our acceptance tests are different. We start docker instances on travis, install dependencies from our metadata.json, execute puppet  code and check if everything   is working. We're simulating our users.

Combination of unit and acceptance tests are, in my opinion, the best way to address/track breaking changes.

This was a rather long explanation and I'm not a native speaker. If this text raises more questions than it answers please feel free to join our IRC channel #voxpupuli on freenode and we can talk about this.